tests/Benchmarks/System.Text.JsonLab/JsonReaderPerf.cs

```diff
@@ -2,70 +2,277 @@
+                    break;
+                case TestCaseType.DeepTree:
+                    _jsonString = JsonStrings.DeepTree;
+                    break;
+                case TestCaseType.BroadTree:
+                    _jsonString = JsonStrings.BroadTree;
+                    break;
+                case TestCaseType.LotsOfNumbers:
+                    _jsonString = JsonStrings.LotsOfNumbers;
+                    break;
+                case TestCaseType.LotsOfStrings:
+                    _jsonString = JsonStrings.LotsOfStrings;
+                    break;
+                case TestCaseType.Json400B:
+                    _jsonString = JsonStrings.Json400B;
+                    break;
+                case TestCaseType.Json4KB:
+                    _jsonString = JsonStrings.Json4KB;
+                    break;
+                case TestCaseType.Json40KB:
+                    _jsonString = JsonStrings.Json40KB;
+                    break;
+                case TestCaseType.Json400KB:
+                    _jsonString = JsonStrings.Json400KB;
+                    break;
+            }
 
-            _stream = new MemoryStream(_data);
-            var enc = Target == EncoderTarget.InvariantUtf8 ? Encoding.UTF8 : Encoding.Unicode;
-            _reader = new StreamReader(_stream, enc, false, 1024, true);
+            if (IsUTF8Encoded)
+            {
+                _dataUtf8 = Encoding.UTF8.GetBytes(_jsonString);
+                _stream = new MemoryStream(_dataUtf8);
+                _reader = new StreamReader(_stream, Encoding.UTF8, false, 1024, true);
+            }
+            else
+            {
+                _dataUtf16 = Encoding.Unicode.GetBytes(_jsonString);
+            }
         }
 
         [Benchmark(Baseline = true)]
-        public void ReaderNewtonsoftStringReader()
+        public void ReaderNewtonsoftReaderEmptyLoop()
         {
-            var json = new Newtonsoft.Json.JsonTextReader(new StringReader(JsonString));
-            while (json.Read()) ;
+            if (IsUTF8Encoded)
+            {
+                _stream.Seek(0, SeekOrigin.Begin);
+                using (var json = new Newtonsoft.Json.JsonTextReader(_reader))
+                    while (json.Read()) ;
+            }
+            else
+            {
+                var json = new Newtonsoft.Json.JsonTextReader(new StringReader(_jsonString));
+                while (json.Read()) ;
+            }
         }
 
         [Benchmark]
-        public void ReaderNewtonsoftStreamReader()
+        public string ReaderNewtonsoftReaderReturnString()
         {
-            _stream.Seek(0, SeekOrigin.Begin);
-            using (var json = new Newtonsoft.Json.JsonTextReader(_reader))
-                while (json.Read()) ;
+            StringBuilder sb = new StringBuilder();
+            if (IsUTF8Encoded)
+            {
+                _stream.Seek(0, SeekOrigin.Begin);
+                using (var json = new Newtonsoft.Json.JsonTextReader(_reader))
```

Comment:
why one of the readers is wrapped with `using` block, and the other is not?

Reply:
Fixed to be consistent.